### PR TITLE
[Android] IndicatorView with DataTemplate (custom image) does not render correctly when ItemsSource change

### LINKED ISSUE #21980: IndicatorView with DataTemplate (custom image) does not render correctly on ItemsSource change (Android & iOS)
### Description

When using an IndicatorView attached to a CarouselView with a DataTemplate specified, changing the ItemsSource of the CarouselView will cause the updated IndicatorView layout to mess up. This happens on both iOS and Android but in different ways.

On iOS it results in scaled images and on Android you end up with lots of trailing white space. See animated gifs below.

| iOS | Android |
| --- | -------- |
| IndicatorView_iOS | IndicatorView_Android

### Steps to Reproduce

1. Run repo project (on Android and iOS)
2. Click Change Source button
3. Observe broken IndicatorView render/layout.

### Link to public reproduction project repository

https://github.com/msivers/Maui-IndicatorView-Issue

### Version with bug

8.0.20 SR4

### Is this a regression from previous behavior?

Yes, this used to work in Xamarin.Forms

### Last version that worked well

Unknown/Other

### Affected platforms

iOS, Android

### Affected platform versions

_No response_

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by NanthiniMahalingam**
### Root Cause

- When changing itemssource for indicator view, indicator size does not updated based on current items source count. 
- When changing items source count mapping, update indicator view template code section does not include to update the indicator view bounds.

### Description of Change

After adding the Update indicator view template in UpdatingIndcatorCount method for mapping the Count property in IndicatorViewHandler android class, indicator view data template does not mess up.

### Issues Fixed
Fixes #21980

Output:
Android Platform

Before changes:

https://github.com/user-attachments/assets/b4006515-35c3-403b-ba76-b6e101f62e59

After changes:

https://github.com/user-attachments/assets/b94d73c2-5a5a-4534-86d9-7459aaab369c

### CODE REVIEW

File: src/Controls/tests/TestCases.HostApp/Issues/Issue21980.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+	public class Issue21980ViewModel : INotifyPropertyChanged
+	{
+		private readonly IReadOnlyList<string> _imageSource1 = new List<string>
+		{
+			"https://xcdn.next.co.uk/Common/Items/Default/Default/ItemImages/AltItemZoom/440022s.jpg",
+			"https://xcdn.next.co.uk/Common/Items/Default/Default/ItemImages/AltItemZoom/440022s2.jpg",
+			"https://xcdn.next.co.uk/Common/Items/Default/Default/ItemImages/AltItemZoom/440022s3.jpg",
+			"https://xcdn.next.co.uk/Common/Items/Default/Default/ItemImages/AltItemZoom/440022s4.jpg",
+			"https://xcdn.next.co.uk/Common/Items/Default/Default/ItemImages/AltItemZoom/440022s5.jpg"
```
Comment (mattleibow) [inline src/Controls/tests/TestCases.HostApp/Issues/Issue21980.cs:25]: Could these rather be a collection of images in the app, or maybe the dotnet bot multiple times. The network on CI is never reliable enough for testing. No idea why everything else will work but when you need the network for a test it stops working.

**Review by mattleibow — COMMENTED** [review]
Just adding a comment about the test and not actually reviewing the code yet.

File: src/Controls/tests/TestCases.HostApp/Issues/Issue21980.cs (hunk 1 above)
Comment (NanthiniMahalingam) [reply src/Controls/tests/TestCases.HostApp/Issues/Issue21980.cs:25]: I will modify the collection of images in the app, or possibly use the dotnet bot image multiple times.

**Review by PureWeen — CHANGES_REQUESTED** [review]
I tested with the users sample and PR [here](https://github.com/dotnet/maui/pull/24776)

And noticed that with just the fix in this PR that the width of the indicator view stays incorrect

image

where as with the other PR the size seems to update

image

Can you review https://github.com/dotnet/maui/pull/24776 and see if those changes makes sense or if you have thoughts on a different approach?

**Comment by NanthiniMahalingam** [comment]
> I tested with the users sample and PR [here](https://github.com/dotnet/maui/pull/24776)
> 
> And noticed that with just the fix in this PR that the width of the indicator view stays incorrect
> 
> image
> 
> where as with the other PR the size seems to update
> 
> image
> 
> Can you review #24776 and see if those changes makes sense or if you have thoughts on a different approach?

The changes added in the other PR is fine.

**Comment by PureWeen** [comment]
> [#24717 (review)](https://github.com/dotnet/maui/pull/24717#pullrequestreview-2326590412)
> 
> The changes added in the other PR (#24776) is fine.

Closing in favor of

https://github.com/dotnet/maui/pull/24776

**Pull request closed without merging**